wfl/fit/ace.py

```diff
@@ -138,50 +208,61 @@ def _check_output_files(file_base, formats):
 
         return results
 
-    if not run_dir.exists():
-        run_dir.mkdir(parents=True)
+    run_dir.mkdir(exist_ok=True, parents=True)
 
-    use_params = dict(params)
-
-    if dry_run:
-        use_params['dry_run'] = None
-
-    # the code below needs to know an unfortunate amount about the inner workings of ace_fit.jl
-    # here we rely on knowledge of the --outfile_base
-    use_params['outfile_base'] = str(ace_file_base)
-    use_params['outfile_format'] = formats
-
-    use_params['key'] = [ [k, ref_property_prefix + v] for k,v in [('E', 'energy'), ('F', 'forces'), ('V', 'virial')] ]
-    use_params['_repeat_key'] = True
+    use_params = deepcopy(ace_fit_params)
 
     # configs need to be in memory so they can be modified with stress -> virial, and safest to
     # have them as a list (rather than using ConfigSet_in.to_memory()) when passing to ase.io.write below
     fitting_configs = list(fitting_configs)
 
-    # calculate virial from stress, since ASE uses stress but ace_fit.jl only knows about virial
-    for at in fitting_configs:
-        if ref_property_prefix + 'stress' in at.info:
-            stress = at.info[ref_property_prefix + 'stress']
-            if len(stress.shape) == 2:
-                # 3x3
-                at.info[ref_property_prefix + 'virial'] = list((-stress * at.get_volume()).ravel())
-            else:
-                # Voigt 6-vector
-                at.info[ref_property_prefix + 'virial'] = list((-voigt_6_to_full_3x3_stress(stress) * at.get_volume()).ravel())
+    fit_cfgs_fname_given = None 
+    if "fname" in use_params["data"].keys():
+        fit_cfgs_fname_given = use_params["data"]["fname"]
 
-    # write to file since that's what script needs as input
-    fitting_configs_filename = str(run_dir / f'fitting_database.{ACE_name}.extxyz')
-    ase.io.write(fitting_configs_filename, fitting_configs)
+    fit_cfgs_fname = str(run_dir / f'fitting_database.{ace_file_base.name}.extxyz')
+    if fit_cfgs_fname_given is not None and fit_cfgs_fname_given.exists():
+        warnings.warn(f"File name for configs ace_params ({fit_cfgs_fname_given}), exists; "
+                      f"Fitting to configs given to ace.fit(), which are in {fit_cfgs_fname}.")
 
-    use_params["atoms_filename"] = fitting_configs_filename
+    ase.io.write(fit_cfgs_fname, fitting_configs)
+    use_params["data"]["fname"] = fit_cfgs_fname
 
-    fitting_line = dict_to_ace_fit_string(use_params)
+    ace_fit_params_filename = ace_file_base.parent / (ace_file_base.name + "_fit_params.json")
+    print(ace_fit_params_filename)
+    with open(ace_fit_params_filename, "w") as f:
+        f.write(json.dumps(use_params))
 
-    cmd = f'{ace_fit_exec} {fitting_line} > {ace_file_base}.stdout 2> {ace_file_base}.stderr'
+    cmd = f"{ace_fit_exec} --fit-params {ace_fit_params_filename} "
```

Comment:
Is it worth it to pass a JSON on the command line, so we don't have to write a temporary file?

Reply:
I would like to write to file, because that makes it more transparent and easier to debug. The fitting conifgs, `.stderr` and `.stdout` are written to file anyway, so I don't think a parameters' file adds any more clutter. Also saves us from escaping special command line characters and `ace_fit.jl` needing to parse that (maybe the last bit isn't too big of a problem, though).

Reply:
OK.